packages/components/src/lib/hooks/useSquiggleRunner.ts

```diff
@@ -51,69 +50,76 @@ function useSetup(sourceId: string | undefined, setup: SetupSettings) {
       : setup.continues ?? defaultContinues;
 
   const project = useMemo(() => {
-    if (setup.type === "project") {
-      return setup.project;
-    } else {
-      const _project = SqProject.create();
-      if (setup.environment) {
-        _project.setEnvironment(setup.environment);
-      }
-      return _project;
+    switch (setup.type) {
+      case "standalone":
+        return SqProject.create({ environment });
+      case "projectFromLinker":
+        return SqProject.create({ environment, linker: setup.linker });
+      case "project":
+        return setup.project;
+      default:
+        throw new Error("Invalid setup type");
     }
-  }, [setup]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []); // The project should not change.
 
   return { sourceId: _sourceId, project, continues };
 }
 
 export function useSquiggleRunner(
   args: SquiggleRunnerArgs
 ): SquiggleRunnerOutput {
-  const [autorunMode, setAutorunMode] = useState(true);
-
-  // const isViewerTabSet = useRef(false);
-  const [viewerTab, setViewerTab] = useState<ViewerTab | undefined>(undefined);
+  const [autorunMode, setAutorunMode] = useState(
+    args.initialAutorunMode ?? true
+  );
 
-  const { sourceId, project, continues } = useSetup(args.sourceId, args.setup);
+  const { sourceId, project, continues } = useSetup(
+    args.sourceId,
+    args.setup,
+    args.environment
+  );
 
-  const [squiggleProjectRun, { rerunSquiggleCode }] = useSquiggleProjectRun({
+  const [squiggleProjectRun, { runSquiggleProject }] = useSquiggleProjectRun({
     sourceId,
     code: args.code,
     project,
     continues,
     autorunMode,
   });
 
-  // Set viewerTab the first time that SqOutputResult is not undefined.
+  // runSquiggleProject changes every time the code changes. That then triggers this, which would call runSquiggleProject to actually run that updated function.
```

Comment:
> runSquiggleProject changes every time the code changes

`useEffect` dependent on function identity is kinda ok when you do it in the same component, but feels weird as a public contract of `useSquiggleProjectRun` hook.

I don't have any better ideas for now, will keep thinking about it.